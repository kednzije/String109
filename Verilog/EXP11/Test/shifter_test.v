`timescale 10ns/10ns

`include "../Shifter.v"

module shifter_test;
    
    reg     [31:0]  a, b;
    wire    [31:0]  c;
    reg             l_or_r, a_or_h;
    Shifter shifter(
        .dataa(a),
        .datab(b),
        .l_or_r(l_or_r),
        .a_or_h(a_or_h),
        .result(c)
    );

    initial begin
        $display("start a test for Module Shifter.");   // 打印开始标记
        $dumpfile("shifter_wave.vcd");                  // 指定记录模拟波形的文件
        $dumpvars(0, shifter_test);                     // 指定记录的模块层级
        #6000 $finish;                                  // 6000个单位时间后结束模拟
    end

    always begin
        a <= -8;
        b <= 2;
        l_or_r <= 0;
        a_or_h <= 0;
        #20;

        l_or_r <= 1;
        a_or_h <= 0;
        #20;

        l_or_r <= 0;
        a_or_h <= 1;
        #20;

        l_or_r <= 1;
        a_or_h <= 1;
        #20;
    end

endmodule //clock_pulse