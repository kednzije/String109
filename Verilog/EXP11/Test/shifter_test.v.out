#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1110-g18392a46)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 8;
:vpi_module "E:\iverilog\lib\ivl\system.vpi";
:vpi_module "E:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "E:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "E:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "E:\iverilog\lib\ivl\va_math.vpi";
S_0000024d505e9d10 .scope module, "shifter_test" "shifter_test" 2 5;
 .timescale -8 -8;
v0000024d505ebef0_0 .var "a", 31 0;
v0000024d505ee960_0 .var "a_or_h", 0 0;
v0000024d505eea00_0 .var "b", 31 0;
v0000024d505eeaa0_0 .net "c", 31 0, v0000024d505ebe50_0;  1 drivers
v0000024d505eeb40_0 .var "l_or_r", 0 0;
S_0000024d505ee7d0 .scope module, "shifter" "Shifter" 2 10, 3 1 0, S_0000024d505e9d10;
 .timescale -8 -8;
    .port_info 0 /INPUT 32 "dataa";
    .port_info 1 /INPUT 32 "datab";
    .port_info 2 /INPUT 1 "l_or_r";
    .port_info 3 /INPUT 1 "a_or_h";
    .port_info 4 /OUTPUT 32 "result";
v0000024d505e7150_0 .net "a_or_h", 0 0, v0000024d505ee960_0;  1 drivers
v0000024d505e6d90_0 .net "dataa", 31 0, v0000024d505ebef0_0;  1 drivers
v0000024d505e9ea0_0 .net "datab", 31 0, v0000024d505eea00_0;  1 drivers
v0000024d505e9f40_0 .net "l_or_r", 0 0, v0000024d505eeb40_0;  1 drivers
v0000024d505ebe50_0 .var "result", 31 0;
E_0000024d50638c80 .event anyedge, v0000024d505e7150_0, v0000024d505e9f40_0, v0000024d505e6d90_0, v0000024d505e9ea0_0;
    .scope S_0000024d505ee7d0;
T_0 ;
    %wait E_0000024d50638c80;
    %load/vec4 v0000024d505e7150_0;
    %load/vec4 v0000024d505e9f40_0;
    %concat/vec4; draw_concat_vec4
    %dup/vec4;
    %pushi/vec4 2, 2, 2;
    %cmp/x;
    %jmp/1 T_0.0, 4;
    %dup/vec4;
    %pushi/vec4 1, 0, 2;
    %cmp/x;
    %jmp/1 T_0.1, 4;
    %dup/vec4;
    %pushi/vec4 3, 0, 2;
    %cmp/x;
    %jmp/1 T_0.2, 4;
    %pushi/vec4 4294967295, 4294967295, 32;
    %store/vec4 v0000024d505ebe50_0, 0, 32;
    %jmp T_0.4;
T_0.0 ;
    %load/vec4 v0000024d505e6d90_0;
    %load/vec4 v0000024d505e9ea0_0;
    %parti/s 5, 0, 2;
    %ix/vec4 4;
    %shiftl 4;
    %store/vec4 v0000024d505ebe50_0, 0, 32;
    %jmp T_0.4;
T_0.1 ;
    %load/vec4 v0000024d505e6d90_0;
    %load/vec4 v0000024d505e9ea0_0;
    %parti/s 5, 0, 2;
    %ix/vec4 4;
    %shiftr 4;
    %load/vec4 v0000024d505e6d90_0;
    %parti/s 1, 31, 6;
    %replicate 32;
    %pushi/vec4 32, 0, 32;
    %load/vec4 v0000024d505e9ea0_0;
    %parti/s 5, 0, 2;
    %pad/u 32;
    %sub;
    %ix/vec4 4;
    %shiftl 4;
    %or;
    %store/vec4 v0000024d505ebe50_0, 0, 32;
    %jmp T_0.4;
T_0.2 ;
    %load/vec4 v0000024d505e6d90_0;
    %load/vec4 v0000024d505e9ea0_0;
    %parti/s 5, 0, 2;
    %ix/vec4 4;
    %shiftr 4;
    %store/vec4 v0000024d505ebe50_0, 0, 32;
    %jmp T_0.4;
T_0.4 ;
    %pop/vec4 1;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_0000024d505e9d10;
T_1 ;
    %vpi_call 2 19 "$display", "start a test for Module Shifter." {0 0 0};
    %vpi_call 2 20 "$dumpfile", "shifter_wave.vcd" {0 0 0};
    %vpi_call 2 21 "$dumpvars", 32'sb00000000000000000000000000000000, S_0000024d505e9d10 {0 0 0};
    %delay 6000, 0;
    %vpi_call 2 22 "$finish" {0 0 0};
    %end;
    .thread T_1;
    .scope S_0000024d505e9d10;
T_2 ;
    %pushi/vec4 4294967288, 0, 32;
    %assign/vec4 v0000024d505ebef0_0, 0;
    %pushi/vec4 2, 0, 32;
    %assign/vec4 v0000024d505eea00_0, 0;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v0000024d505eeb40_0, 0;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v0000024d505ee960_0, 0;
    %delay 20, 0;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0000024d505eeb40_0, 0;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v0000024d505ee960_0, 0;
    %delay 20, 0;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v0000024d505eeb40_0, 0;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0000024d505ee960_0, 0;
    %delay 20, 0;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0000024d505eeb40_0, 0;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0000024d505ee960_0, 0;
    %delay 20, 0;
    %jmp T_2;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "shifter_test.v";
    "./../Shifter.v";
